Utils/github_workflow_scripts/handle_external_pr.py

```diff
@@ -253,35 +255,124 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
             return True
+    except Exception as er:
+        print(f"The pack is not TIM: {er}")
+        return False
+    return False
+
+
+def check_files_of_pr_manually(pr_files: list[str]) -> bool:
+    """
+    If the checkout of the branch has failed, this function will go over the files and check whether the contribution
+    need to be reviewed by TIM owner
+
+    Arguments:
+        - `pr_files`: ``List[str]``: The list of files changed in the Pull Request. Will be used to determine
+        whether a security engineer is required for the review.
+
+    Returns: `bool` whether a security engineer should be assigned
+    """
+    pack_dirs_to_check = packs_to_check_in_pr(pr_files)
+    pack_metadata_list = get_metadata(pack_dirs_to_check)
+    for file in pr_files:
+        if "yml" in file and "Integrations" in file:
+            content_yml = get_yaml(file_path=file)
+            is_feed = content_yml.get("script").get("feed", "False")
+            print(f'Is it a feed: {is_feed}')
+            if is_feed:
+                return True
+    for pack_metadata in pack_metadata_list:
+        print(f'the metadata is: {pack_metadata}')
+        tags = pack_metadata.get("tags")
+        categories = pack_metadata.get("categories")
+        if TIM_TAGS in tags or TIM_CATEGORIES in categories:  # type: ignore
+            return True
+    return False
+
+
+def check_new_pack_metadata(pr_files: list[str], external_pr_branch: str, repo_name: str) -> bool:
+    """
+    Checks if tim reviewer needed, if the pack is new and not part of Master.
+    First the remote branch is going to be checked out and then verified for the data
+
+    Arguments:
+        - `pr_files`: ``List[str]``: The list of files changed in the Pull Request. Will be used to determine
+        whether a security engineer is required for the review.
+        - 'external_pr_branch': str : name of branch to checkout
+        - 'repo_name': str : name of repository
+
+    Returns: `bool` whether a security engineer should be assigned
+    """
+    try:
+        fork_owner = os.getenv('GITHUB_ACTOR')
+        with Checkout(
+            repo=Repo(Path().cwd(), search_parent_directories=True),
+            branch_to_checkout=external_pr_branch,
+            # in marketplace contributions the name of the owner should be xsoar-contrib
+            fork_owner=fork_owner if fork_owner != 'xsoar-bot' else 'xsoar-contrib',
+            repo_name=repo_name
+        ):
+            for file in pr_files:
+                if 'CONTRIBUTORS.json' in file or 'Author_image' in file or 'README.md' in file or ".pack-ignore" in file:
+                    continue
+                content_object = BaseContent.from_path(CONTENT_PATH / file)
+                is_tim_needed = check_if_pack_or_integration_is_feed(content_object)
+                if is_tim_needed:
+                    return True
+    except Exception as er:
+        print(f"couldn't checkout branch to get metadata, error is {er}")
```

Comment:
Why would this happen?

Reply:
This will happen if we don't have permissions to checkout remote branch. this happens sometimes.
depends on the remote settings, now all of them allow to checkout their branch